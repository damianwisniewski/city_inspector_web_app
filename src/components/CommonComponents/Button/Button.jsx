import React from 'react'
import PropTypes from 'prop-types'

import './Button.scss'

const Button = ({ color, className, type, children, category, onClick, ...props }) => (
	<button
		{...props}
		type={type}
		onClick={onClick}
		className={`${category === 'text' ? 'text-button' : 'button'} ${color} ${className}`}
	>
		{children}
	</button>
)

Button.defaultProps = {
	color: 'white',
	children: '',
	className: '',
	type: 'button',
	category: 'button',
	onClick: () => {},
}

Button.propTypes = {
	color: PropTypes.oneOf(['white', 'blue', 'green', 'red']),
	children: PropTypes.node,
	className: PropTypes.string,
	onClick: PropTypes.func,
	type: PropTypes.string,
	category: PropTypes.oneOf(['button', 'text']),
}

export default Button
