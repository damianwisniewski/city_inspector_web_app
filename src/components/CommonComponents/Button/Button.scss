@import '../../../assets/styles/index';

$buttonColors: (
	red: $red,
	green: $green,
	white: $white,
	blue: $blue,
);

.custom-button {
	font-family: $secondary_font;
	font-size: 14px;
	font-weight: 900;

	padding: 10px 20px;
	margin: 0 10px;

	box-shadow: $small_shadow;
	border-radius: 5px;

	border: none;
	outline: none;

	transition: background-color 0.1s ease-out;

	cursor: pointer;

	// Creates separate styles for each button color
	@each $name, $color in $buttonColors {
		$font-color: $white;

		@if $name == white {
			$font-color: $dark_gray;
		}

		&.#{$name} {
			background-color: $color;
			color: $font-color;

			&:active {
				background-color: darken($color: $color, $amount: 10%);
			}
		}
	}
}
