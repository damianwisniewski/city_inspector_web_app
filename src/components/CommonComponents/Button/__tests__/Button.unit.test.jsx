import React from 'react'
import Button from '../Button'
import { configure, shallow, ShallowWrapper } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

configure({ adapter: new Adapter() })

describe('<Button />', () => {
	/**
	 * @type {ShallowWrapper}
	 */
	let ButtonComponent
	const ButtonText = 'ExampleText'

	beforeEach(() => {
		ButtonComponent = shallow(<Button>{ButtonText}</Button>)
	})

	test('should render correctly without any props', () => {
		expect(toJson(ButtonComponent)).toMatchSnapshot()
	})

	test('should render correctly with some additionally props', () => {
		ButtonComponent.setProps({
			color: 'green',
			className: 'test-class',
			type: 'submit',
			category: 'text',
			onClick: () => {},
		})
		expect(toJson(ButtonComponent)).toMatchSnapshot()
	})

	test('should call function passed in "onClick" props, on action click', () => {
		const spy = jest.fn()
		ButtonComponent.setProps({ onClick: spy })

		ButtonComponent.simulate('click')
		expect(spy).toBeCalledTimes(1)
	})
})
