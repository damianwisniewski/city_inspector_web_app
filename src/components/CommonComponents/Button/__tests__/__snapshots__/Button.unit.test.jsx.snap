// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly with some additionally props 1`] = `
<button
  className="text-button green test-class"
  onClick={[Function]}
  type="submit"
>
  ExampleText
</button>
`;

exports[`<Button /> should render correctly without any props 1`] = `
<button
  className="button white "
  onClick={[Function]}
  type="button"
>
  ExampleText
</button>
`;
